### PR TITLE
Fix for bug where cards are marked as saved before save completes

### PR DESCRIPTION
**Pull request by habdelra**
This PR fixes a bug where cards are being marked as saved before the save happens. This also fixes an issue where a card is being saved when exiting edit mode even though the card never changed.

### CODE REVIEW

File: packages/host/app/components/operator-mode/stack-item.gts [hunk 1]
```diff
@@ -320,12 +321,15 @@ export default class OperatorModeStackItem extends Component<Signature> {
...
-    this.args.publicAPI.saveCard(this.card, false);
-    this.isSaving = false;
-    this.lastSaved = Date.now();
-    this.calculateLastSavedMsg();
+    this.args.publicAPI.saveCard(this.card, false, () => {
+      this.hasUnsavedChanges = false;
+      this.isSaving = false;
+      this.lastSaved = Date.now();
+      this.calculateLastSavedMsg();
+    });
```
Comment (lukemelia) [inline packages/host/app/components/operator-mode/stack-item.gts]: Why introduce an after callback instead of making saveCard async?

**Comment by lukemelia** [comment]
It should be possible to assert intermediate state even with test waiters. Avoid "await"-ing the pervious operation then add a `waitFor` or `waitUntil` so that you can assert the expected state occurs.

File: packages/host/app/components/operator-mode/stack-item.gts (hunk 1 above)
Comment (habdelra) [reply packages/host/app/components/operator-mode/stack-item.gts]: good point. i was trying to be as "light touch" as possible with the API. but really there is no need to not be async

Comment (habdelra) [reply packages/host/app/components/operator-mode/stack-item.gts]: actually wait-- there is a reason.  I'm intentionally not awaiting an EC perform. When you do await an EC perform that actually casts it into an uncancellable promise. You should never `await` an EC perform outside of a task (inside of a task an `await` is actually syntactic sugar for a `yield`). more info about this is here: https://ember-concurrency.com/docs/task-cancelation-help. I can rewrite this to still have an async interface, but we'll have to use a deferred and be _very_ careful to not await the EC perform (and document that as such)

File: packages/host/app/components/operator-mode/interact-submode.gts [hunk 2]
```diff
@@ -218,9 +218,17 @@ export default class InteractSubmode extends Component<Signature> {
...
-        here.save.perform(item, dismissItem);
+        // WARNING: never await an ember concurrency perform outside of a task.
+        // Inside of a task, the `await` is actually just syntactic sugar for a
+        // `yield`. But if you await `perform()` outside of a task, that will cast
+        // the the task to an uncancellable promise which defeats the point of using
+        // ember concurrency.
+        // https://ember-concurrency.com/docs/task-cancelation-help
+        let deferred = new Deferred<void>();
+        here.save.perform(item, dismissItem, deferred);
+        await deferred.promise;
```
Comment (habdelra) [inline packages/host/app/components/operator-mode/interact-submode.gts:231]: @lukemelia this is a bit more complex implementation in order to tap dance around awaiting the perform and have an async interface. do you think the tradeoff is worth it? (make sure to note how the `this.save` has changed too)

**Pull request merged**